ovos_core/intent_services/padatious_service.py

```diff
@@ -72,84 +74,72 @@ class PadatiousMatcher:
         # call flatten in case someone is sending the old style list of tuples
         utterances = flatten_list(utterances)
-        if not self.has_result:
-            lang = lang or self.service.lang
-            LOG.debug(f'Padatious Matching confidence > {limit}')
-            if _pd:
-                LOG.info(f"Using legacy Padatious module")
-                padatious_intent = self._legacy_padatious_match(utterances,
-                                                                lang)
-            else:
-                padatious_intent = self.service.threaded_calc_intent(utterances,
-                                                                     lang)
-            if padatious_intent:
-                skill_id = padatious_intent.name.split(':')[0]
-                self.ret = ovos_core.intent_services.IntentMatch(
-                    'Padatious', padatious_intent.name,
-                    padatious_intent.matches, skill_id)
-                self.conf = padatious_intent.conf
-            self.has_result = True
-        if self.conf and self.conf > limit:
-            return self.ret
-
-    def match_high(self, utterances, lang=None, __=None):
+        lang = lang or self.service.lang
+        return self.service.calc_intent(utterances, lang)
+
+    @staticmethod
+    def _convert_intent(padatious_intent: PadatiousIntent, limit: int):
+        """ helper method to ensure service returns correct type: IntentMatch named tuple"""
+        if padatious_intent.conf > limit:
+            skill_id = padatious_intent.name.split(':')[0]
+            return ovos_core.intent_services.IntentMatch(
+                'Padatious', padatious_intent.name,
+                padatious_intent.matches, skill_id)
+
+    def match_high(self, utterances, lang=None, message=None):
         """Intent matcher for high confidence.
 
         Args:
             utterances (list of tuples): Utterances to parse, originals paired
                                          with optional normalized version.
         """
-        return self._match_level(utterances, self.service.conf_high, lang)
+        if self.service.is_regex_only:
+            LOG.debug(f'Padacioso Matching confidence > {self.service.conf_high}')
+        else:
+            LOG.debug(f'Padatious Matching confidence > {self.service.conf_high}')
+        padatious_intent = self._match_cached(utterances, lang)
+        if padatious_intent:
+            return self._convert_intent(padatious_intent, self.service.conf_high)
 
-    def match_medium(self, utterances, lang=None, __=None):
+    def match_medium(self, utterances, lang=None, message=None):
         """Intent matcher for medium confidence.
 
         Args:
             utterances (list of tuples): Utterances to parse, originals paired
                                          with optional normalized version.
         """
-        return self._match_level(utterances, self.service.conf_med, lang)
+        if self.service.is_regex_only:
```

Comment:
Can this be consolidated in `self._match_level` instead of duplicating across methods?

Reply:
self._match_level no longer exists, it was split out in order to allow the usage of lru_cache, i had explicitly to remove this argument from that method